src/amo/reducers/home.js

```diff
@@ -127,11 +131,12 @@ const reducer = (
         ...state,
         collections: collections.map((collection) => {
           if (collection) {
-            return collection.results
-              .slice(0, LANDING_PAGE_ADDON_COUNT)
-              .map((item) => {
-                return createInternalAddon(item.addon);
-              });
+            const sliceEnd = isTheme(collection.results[0].addon.type)
```

Comment:
not sure if this is ideal way to figure out but it seems to work so maybe this is okay..?

Reply:
Hmm, yeah, this could be an issue if we had mixed content in a collection, but we were just told today that that wouldn't happen, so I guess it's okay to do. There is a problem though, that you could have an empty collection, in which case `collection.results[0]` would be `undefined` and this code would throw an exception, so you need to check for that. I think you could just check for `collection.results.length` here and not do any other processing if it's empty, and let it fall through to the `return null` at line 141.